biocypher_metta/adapters/endb_adapter.py

```diff
@@ -0,0 +1,174 @@
+
+        self.source = 'ENdb'
+        self.version = ''
+        self.source_url = 'https://bio.liclab.net/ENdb/Download.php'
+
+        super(ENdbAdapter, self).__init__(write_properties, add_provenance)
+
+    def parse_info_metadata(self, info):
+        parsed_info = info.replace('"', '').split(',')
+        return parsed_info
+    
+    def handle_target_genes(self, genes):
+        parsed_genes = self.parse_info_metadata(genes)
+        result = []
+        for gene in parsed_genes:
+            if gene in self.hgnc_to_ensembl_map:
+                result.append(self.hgnc_to_ensembl_map[gene])
+            elif gene.replace('-', '') in self.hgnc_to_ensembl_map:
+                result.append(self.hgnc_to_ensembl_map[gene.replace('-', '')])
+            elif gene[:-1] in self.hgnc_to_ensembl_map:
+                result.append(self.hgnc_to_ensembl_map[gene[:-1]])
+        if not result and parsed_genes:
+            result.append(parsed_genes[0])
+            
+        return result
+    
+    def convert_to_hg38(self, chr,  pos):
+        try:
+            chr_no = chr.replace('chr', '').replace('ch', '')
+            converted = self.genome_reference_converter.query(chr_no, pos)[0][1]
+            return converted
+        except:
+            return False
+    
+    def get_nodes(self):
+        self.temp = []
+        with gzip.open(self.filepath, 'rt') as f:
+            next(f) # Skip the header row
+            for line in f:
+                cur_data = line.strip().split(self.delimiter)
+                first_val = cur_data[0]
+                
+                if first_val.startswith('E_'):
+                    if self.temp:
+                        data = self.temp.copy()
+                    else:
+                        self.temp = cur_data.copy()
+                        continue
+                else:
+                    curr = cur_data.pop(0)
+                    self.temp[-1] = self.temp[-1] + ' ' + curr
+                    self.temp = self.temp + cur_data
+                    continue
+
+                self.temp = cur_data
+                species = data[ENdbAdapter.INDEX['species']]
+                enhancer_type = data[ENdbAdapter.INDEX['enhancer_type']]
+                
+                if species != 'Human':
+                    continue
+                if enhancer_type.lower().replace('-',' ') != self.type:
+                    continue
+                enhancer_id = data[ENdbAdapter.INDEX['enhancer_id']]
+                chr = data[ENdbAdapter.INDEX['chr']]
+                start_position = self.convert_to_hg38(chr, int(data[ENdbAdapter.INDEX['coord_start']]))
+                end_position = self.convert_to_hg38(chr, int(data[ENdbAdapter.INDEX['coord_end']]))
+                if not start_position or not end_position:
+                    continue
+                tf_name = data[ENdbAdapter.INDEX['tf_name']]
+                snp_id = data[ENdbAdapter.INDEX['snp_id']]
```

Comment:
Perhaps this should be an edge such as `enhancer to snp association`? If it takes much time to include do this add it as a TODO comment for now and you can raise another PR for it